model/riscv_sys_regs.sail

```diff
@@ -160,15 +160,16 @@ function legalize_misa(m : Misa, v : xlenbits) -> Misa = {
   then { /* Handle modifications to C. */
          let  v = Mk_Misa(v);
          /* Suppress changing C if nextPC would become misaligned or an extension vetoes or C was disabled at boot (i.e. not supported). */
-         let m =
-           if   (v.C() == 0b0 & (nextPC[1] == bitone | ext_veto_disable_C())) | ~(sys_enable_rvc())
-           then m
-           else update_C(m, v.C());
-         /* Handle updates for F/D. */
-         if   ~(sys_enable_fdext()) | (v.D() == 0b1 & v.F() == 0b0) | (v.C() == 0b0 & (nextPC[1] == bitone))
-         then m
-         else update_D(update_F(m, v.F()), v.D())
-       }
+         if   v.C() == 0b0 & (nextPC[1] == bitone | ext_veto_disable_C())
+	then m
+	else {
+	    	let m' = if ~(sys_enable_rvc()) then m else update_C(m, v.C());
+	    	  /* Handle updates for F/D. */
+		 if   ~(sys_enable_fdext()) | (v.D() == 0b1 & v.F() == 0b0)
+		 then m
```

Comment:
Either these references need to be m' or you should just use `let m =` above to create a new binding that shadows the old one (like this function already does with `v`). Otherwise the misa.C update gets lost.